packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -229,200 +487,122 @@ const npmInstall = (outputDir, {verbose}) => {
     }
 }
 
-// Validations
-const validProjectName = (s) => {
-    const regex = new RegExp(`^[a-zA-Z0-9-\\s]{1,${PROJECT_ID_MAX_LENGTH}}$`)
-    return regex.test(s) || 'Value can only contain letters, numbers, space and hyphens.'
-}
+/**
+ * Execute and copy the handlebars template to the output directory using
+ * the provided context object. If the file isn't a template, simply copy
+ * it to the destination.
+ *
+ * @param {string} inputFile
+ * @param {string} outputDir
+ * @param {Object} context
+ */
+const processTemplate = (relFile, inputDir, outputDir, context) => {
+    const inputFile = p.join(inputDir, relFile)
+    const outputFile = p.join(outputDir, relFile)
+    const destDir = outputFile.split(p.sep).slice(0, -1).join(p.sep)
+
+    // Create folder if we are doing a deep copy
+    if (destDir) {
+        fs.mkdirSync(destDir, {recursive: true})
+    }
 
-const validUrl = (s) => {
-    try {
-        new URL(s)
-        return true
-    } catch (err) {
-        return 'Value must be an absolute URL'
+    if (inputFile.endsWith('.hbs')) {
+        const template = sh.cat(inputFile).stdout
+        fs.writeFileSync(outputFile.replace('.hbs', ''), Handlebars.compile(template)(context))
+    } else {
+        fs.copyFileSync(inputFile, outputFile)
     }
 }
 
-const validSiteId = (s) =>
-    /^[a-z0-9_-]+$/i.test(s) || 'Valid characters are alphanumeric, hyphen, or underscore'
+/**
+ * This function does the bulk of the project generation given the project config
+ * object and the answers returned from the survey process.
+ *
+ * @param {*} preset
+ * @param {*} answers
+ * @param {*} param2
+ */
+const runGenerator = (context, {outputDir, verbose}) => {
+    const {answers, preset} = context
+    const {templateSource} = preset
+    const {extend = false} = answers.project
 
-// To see definitions for Commerce API configuration values, go to
-// https://developer.salesforce.com/docs/commerce/commerce-api/guide/commerce-api-configuration-values.
-const defaultCommerceAPIError =
-    'Invalid format. Use docs to find more information about valid configurations: https://developer.salesforce.com/docs/commerce/commerce-api/guide/commerce-api-configuration-values'
-const validShortCode = (s) => /(^[0-9A-Z]{8}$)/i.test(s) || defaultCommerceAPIError
-const validClientId = (s) =>
-    /(^[0-9A-Z]{8}-[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{12}$)/i.test(s) ||
-    s === 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' ||
-    defaultCommerceAPIError
-const validOrganizationId = (s) =>
-    /^(f_ecom)_([A-Z]{4})_(prd|stg|dev|[0-9]{3}|s[0-9]{2})$/i.test(s) || defaultCommerceAPIError
+    // Check if the output directory doesn't already exist.
+    checkOutputDir(outputDir)
```

Comment:
For better experience, how about we check for the output dir at a much earlier step? 

Otherwise, currently it's possible for a user to enter all of the answers and then find out at the end that the output dir already existed. They later would have to repeat the answers.

Reply:
I don't think I've changed how/when that question is asked from the existing generator. Unless I did I've prefer to follow up since it's not super related to generator and template extensibility.